### ISSUE TITLE
API /certificates endpoint only returns up to 100 SNIs per certificate

### ISSUE THREAD

**Issue by ionosphere80** [Kong/kong#3718]
### Summary

The API /certificates endpoint only returns up to 100 SNIs per certificate. Pagination does not affect output.

### Steps To Reproduce

1. Add a certificate.

    ```
    $ curl -X POST -H "Content-Type: application/json" --data @cert-and-key.json http://kong:8001/certificates
    {
      "id": "6ef44500-e914-4148-9341-ee07ecd9ab8e"
    }
    ```
  
2. Add over 100 arbitrary SNIs to the certificate. The `test-110-snis.json` file contains 110 arbitrary SNIs.

    ```    
    $ curl -X PATCH -H "Content-Type: application/json" --data @test-110-snis.json http://kong:8001/certificates/6ef44500-e914-4148-9341-ee07ecd9ab8e
    ```

3. Read the certificate via certificates/id and see only 100 of the arbitrary SNIs. The missing SNIs are random, but additional reads consistently lack the same SNIs.

    ```
    $ curl http://kong:8001/certificates/6ef44500-e914-4148-9341-ee07ecd9ab8e
    {
      "snis": [
        (only 100 entries present)
      ]
    }
    ```

4. Read one of the missing SNIs via certificates/name and it appears present. For example, `sni12`.

    ```
    $ curl http://kong:8001/certificates/sni12
    {
      "id": "6ef44500-e914-4148-9341-ee07ecd9ab8e"
    }
    ```

### Additional Details & Logs

- Kong version 0.14.0
- Kong configuration

```
{
  "plugins": {
    "enabled_in_cluster": [
      "cors",
      "tcp-log",
      "correlation-id",
      "request-termination",
      "response-transformer",
      "acl",
      "jwt"
    ],
    "available_on_server": {
      "response-transformer": true,
      "oauth2": true,
      "acl": true,
      "correlation-id": true,
      "pre-function": true,
      "jwt": true,
      "cors": true,
      "ip-restriction": true,
      "basic-auth": true,
      "key-auth": true,
      "rate-limiting": true,
      "request-transformer": true,
      "http-log": true,
      "file-log": true,
      "hmac-auth": true,
      "ldap-auth": true,
      "datadog": true,
      "tcp-log": true,
      "zipkin": true,
      "post-function": true,
      "request-size-limiting": true,
      "bot-detection": true,
      "syslog": true,
      "loggly": true,
      "azure-functions": true,
      "udp-log": true,
      "response-ratelimiting": true,
      "aws-lambda": true,
      "statsd": true,
      "prometheus": true,
      "request-termination": true
    }
  },
  "tagline": "Welcome to kong",
  "configuration": {
    "plugins": [
      "bundled"
    ],
    "admin_listen": [
      "0.0.0.0:8001"
    ],
    "lua_ssl_verify_depth": 1,
    "trusted_ips": [
      "0.0.0.0/0",
      "::/0"
    ],
    "prefix": "/usr/local/kong",
    "loaded_plugins": {
      "response-transformer": true,
      "request-termination": true,
      "prometheus": true,
      "ip-restriction": true,
      "pre-function": true,
      "jwt": true,
      "cors": true,
      "statsd": true,
      "basic-auth": true,
      "key-auth": true,
      "ldap-auth": true,
      "aws-lambda": true,
      "http-log": true,
      "response-ratelimiting": true,
      "hmac-auth": true,
      "request-size-limiting": true,
      "datadog": true,
      "tcp-log": true,
      "zipkin": true,
      "post-function": true,
      "bot-detection": true,
      "acl": true,
      "loggly": true,
      "syslog": true,
      "azure-functions": true,
      "udp-log": true,
      "file-log": true,
      "request-transformer": true,
      "correlation-id": true,
      "rate-limiting": true,
      "oauth2": true
    },
    "cassandra_username": "kong",
    "admin_ssl_cert_csr_default": "/usr/local/kong/ssl/admin-kong-default.csr",
    "ssl_cert_key": "/local/blah.key",
    "dns_resolver": [
      "10.60.11.55"
    ],
    "pg_user": "kong_sandbox",
    "mem_cache_size": "128m",
    "cassandra_data_centers": [
      "dc1:2",
      "dc2:3"
    ],
    "nginx_admin_directives": {},
    "custom_plugins": {},
    "pg_host": "blah",
    "nginx_acc_logs": "/usr/local/kong/logs/access.log",
    "proxy_listen": [
      "0.0.0.0:8000",
      "0.0.0.0:8443 ssl proxy_protocol",
      "0.0.0.0:9443 ssl"
    ],
    "client_ssl_cert_default": "/usr/local/kong/ssl/kong-default.crt",
    "ssl_cert_key_default": "/usr/local/kong/ssl/kong-default.key",
    "dns_no_sync": false,
    "db_update_propagation": 0,
    "nginx_err_logs": "/usr/local/kong/logs/error.log",
    "cassandra_port": 9042,
    "dns_order": [
      "LAST",
      "SRV",
      "A",
      "CNAME"
    ],
    "dns_error_ttl": 1,
    "headers": [
      "server_tokens",
      "latency_tokens"
    ],
    "dns_stale_ttl": 4,
    "nginx_optimizations": true,
    "database": "postgres",
    "pg_database": "kong_blah",
    "nginx_worker_processes": "auto",
    "lua_package_cpath": "",
    "admin_acc_logs": "/usr/local/kong/logs/admin_access.log",
    "lua_package_path": "./?.lua;./?/init.lua;",
    "nginx_pid": "/usr/local/kong/pids/nginx.pid",
    "upstream_keepalive": 60,
    "cassandra_contact_points": [
      "127.0.0.1"
    ],
    "client_ssl_cert_csr_default": "/usr/local/kong/ssl/kong-default.csr",
    "proxy_listeners": [
      {
        "ssl": false,
        "ip": "0.0.0.0",
        "proxy_protocol": false,
        "port": 8000,
        "http2": false,
        "listener": "0.0.0.0:8000"
      },
      {
        "ssl": true,
        "ip": "0.0.0.0",
        "proxy_protocol": true,
        "port": 8443,
        "http2": false,
        "listener": "0.0.0.0:8443 ssl proxy_protocol"
      },
      {
        "ssl": true,
        "ip": "0.0.0.0",
        "proxy_protocol": false,
        "port": 9443,
        "http2": false,
        "listener": "0.0.0.0:9443 ssl"
      }
    ],
    "proxy_ssl_enabled": true,
    "admin_access_log": "/dev/stdout",
    "pg_password": "******",
    "enabled_headers": {
      "latency_tokens": true,
      "X-Kong-Proxy-Latency": true,
      "Via": true,
      "server_tokens": true,
      "Server": true,
      "X-Kong-Upstream-Latency": true,
      "X-Kong-Upstream-Status": false
    },
    "cassandra_ssl": false,
    "ssl_cert_csr_default": "/usr/local/kong/ssl/kong-default.csr",
    "db_resurrect_ttl": 30,
    "client_max_body_size": "0",
    "cassandra_consistency": "ONE",
    "db_cache_ttl": 0,
    "admin_error_log": "/dev/stderr",
    "pg_ssl_verify": false,
    "dns_not_found_ttl": 30,
    "pg_ssl": false,
    "client_ssl": false,
    "db_update_frequency": 5,
    "cassandra_repl_strategy": "SimpleStrategy",
    "nginx_kong_conf": "/usr/local/kong/nginx-kong.conf",
    "cassandra_repl_factor": 1,
    "nginx_http_directives": [
      {
        "value": "prometheus_metrics 5m",
        "name": "lua_shared_dict"
      }
    ],
    "error_default_type": "text/plain",
    "kong_env": "/usr/local/kong/.kong_env",
    "cassandra_schema_consensus_timeout": 10000,
    "dns_hostsfile": "/etc/hosts",
    "admin_listeners": [
      {
        "ssl": false,
        "ip": "0.0.0.0",
        "proxy_protocol": false,
        "port": 8001,
        "http2": false,
        "listener": "0.0.0.0:8001"
      }
    ],
    "real_ip_header": "proxy_protocol",
    "ssl_cert": "/local/blah.com.crt",
    "proxy_access_log": "off",
    "admin_ssl_cert_key_default": "/usr/local/kong/ssl/admin-kong-default.key",
    "cassandra_ssl_verify": false,
    "cassandra_lb_policy": "RoundRobin",
    "ssl_cipher_suite": "modern",
    "real_ip_recursive": "on",
    "proxy_error_log": "/dev/stderr",
    "client_ssl_cert_key_default": "/usr/local/kong/ssl/kong-default.key",
    "nginx_daemon": "off",
    "anonymous_reports": true,
    "cassandra_timeout": 5000,
    "nginx_proxy_directives": {},
    "pg_port": 5432,
    "log_level": "notice",
    "client_body_buffer_size": "5m",
    "ssl_ciphers": "ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA256",
    "lua_socket_pool_size": 30,
    "admin_ssl_cert_default": "/usr/local/kong/ssl/admin-kong-default.crt",
    "cassandra_keyspace": "kong",
    "ssl_cert_default": "/usr/local/kong/ssl/kong-default.crt",
    "nginx_conf": "/usr/local/kong/nginx.conf",
    "admin_ssl_enabled": false
  },
  "version": "0.14.0",
  "node_id": "5007e753-ead6-43ce-a272-b0fc4f08a3f8",
  "lua_version": "LuaJIT 2.1.0-beta3",
  "prng_seeds": {
    "pid: 26": 194226191831
  },
  "timers": {
    "pending": 5,
    "running": 0
  },
  "hostname": "blah"
}
```

- Using Docker container `0.14.0-alpine` on Ubuntu 16.04.

**Comment by ionosphere80** [comment]
Adding to this issue, apparently each certificate can only handle 100 SNIs. Adding more SNIs causes Kong to remove existing SNIs for the certificate, seemingly at random, to limit the quantity of SNIs to 100. The previous version, 0.13.1, does not contain this limitation. This limitation effectively breaks my environment, so I must revert to version 0.13.1.

**Comment by thibaultcha** [comment]
This should be fixed now that #3722 - thank you for the report @ionosphere80!
